### ISSUE TITLE
Proposal: Remove Nix support

### ISSUE THREAD

**Issue by paperdev-code** [packwiz/packwiz#332]
I was pleasantly surprised to see this repository had a flake, however, it did not work as expected (#297).

I read through the discussion on #96 and #130 and, realized Nix wasn't really something that should have first class support for a few reasons;

1. The main developer does not use Nix
2. There are no other packaging solutions in this repository (unless you count `go install`).

Of particular note, this question was left entirely unanswered;

> I have no idea what these files do... isn't this something that can be maintained externally?
\- @comp500 

Yes, they can, and it can actually be beneficial for a few reasons;

* The developer does not need to be aware or see these files.
* This repository can be added as a flake input, which is very easy to update.
* This Repository can focus on packwiz only.

I propose the flake be removed from the repository, and move such functionality to either `packwiz/packwiz-flake`, or a third party altogether. A similar approach was taken by the Neovim developers early this year. Such efforts have moved into the `nix-community` org.

Additional developments could also be done there to use `packwiz` declaratively through Nix, that might be neat as well.

In the meantime, if anyone needs to have this work for their _devShell_, here's my workaround;

```nix
{
  description = "A very basic flake";

  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
    systems.url = "github:nix-systems/default";
    packwiz.url = "github:packwiz/packwiz";
    packwiz.flake = false;
  };

  outputs =
    inputs@{ self, nixpkgs, ... }:
    let
      eachSystem =
        f: nixpkgs.lib.genAttrs (import inputs.systems) (system: f nixpkgs.legacyPackages.${system});
    in
    {
      packages = eachSystem (pkgs: {
        packwiz = pkgs.buildGoModule {
          name = "packwiz";
          src = inputs.packwiz;
          vendorHash = "sha256-krdrLQHM///dtdlfEhvSUDV2QljvxFc2ouMVQVhN7A0=";
        };
      });

      devShells = eachSystem (pkgs: {
        default = pkgs.mkShell {
          packages = [
            pkgs.jdk
            self.packages.${pkgs.system}.packwiz
          ];
        };
      });
    };
}
```

**Comment by Silver-Golden** [comment]
As I mentioned on Discord, I am personally not in favor of complete removal of teh flake but I do agree that using it for packaging is not the way to go.  

What I would be in favor of is stripping much of it out and using it purely for teh devshell.  
This would turn it into a tool that devs (like myself) could use via a single command ``nix develop`` and a dev enviroment is setup with ``go`` and any other packages needed to work on ``packwiz``.  
I personally use these so I dont have to worry about having $Language installed globally on my system, and works wonders if I need to sue different versions of $Language for different projects.

The rough flake.nix that I hacked together to work on packwiz is below, far from perfect but did the job.

```nix
{
  description = "Name Here";
  inputs = {
    nixpkgs.url = "nixpkgs/nixos-unstable";
    utils.url = "github:numtide/flake-utils";

    gomod2nix = {
      url = "github:tweag/gomod2nix";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.utils.follows = "utils";
    };

  };

  outputs = {
    self,
    nixpkgs,
    utils,
    gomod2nix
  }:
    utils.lib.eachDefaultSystem (
      system: let
        pkgs = import nixpkgs {
          inherit system;
          overlays = [ gomod2nix.overlays.default ];
        };

        packages = with pkgs; [
           go
           gopls
           gotools
           go-tools
           gomod2nix.packages.${system}.default
           sqlite-interactive
         ];
      in rec {
        # `nix develop`
        devShells.default = pkgs.mkShell {
          buildInputs = packages;
        };
      }
    );
}
```

**Comment by paperdev-code** [comment]
> I do agree that using it for packaging is not the way to go.

I disagree, I think using a flake for packaging is perfectly acceptable. I just don't think this is the right place for it because it's entirely a third party effort which does not need to be in this repository to work.

A flake used for development, logically, should also be able to build the software.

**Comment by xieve** [comment]
Additionally, packwiz is [available via nixpkgs](https://github.com/NixOS/nixpkgs/blob/85dbfc7aaf52ecb755f87e577ddbe6dbbdbc1054/pkgs/by-name/pa/packwiz/package.nix)
